### ISSUE TITLE
Experiments on Qwen3-base models

### ISSUE THREAD

**Issue by 1229095296** [Xuekai-Zhu/FlowRL#8]
Dear Xuekai-Zhu
 Have you ever tried training Qwen3-base models on this code? I use same bash for Qwen2.5-7B and Qwen3-1.7B. I found that 7B takes only 9 minutes while 1.7B takes 48 minutes. I wonder if there is some speed-up methods designed for Qwen2.5?
 Here is my training bash:(vllm 0.8.5)

PRETRAINED_MODEL=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/github.com/volcengine/verl.git/huggingface.co/Qwen/Qwen3-1.7B
n_nodes=1
n_gpus_per_node=8
tensor_model_parallel_size=4
save_freq=50

dapo_train_path=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/flowrl/FlowRL-main/data/dapo-math-17k.parquet
r1_test_path=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/flowrl/FlowRL-main/data/validation.parquet

experiment_name="flowrl_qwen_1.7b_math"
max_prompt_length=2048
max_response_length=8192
OUTPUT_DIR=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/Verl_flow/verl-0.5.0/ckpt/$experiment_name

set -x

python3 -m verl.trainer.main_ppo \
    algorithm.adv_estimator=grpo \
    data.train_files=$dapo_train_path \
    data.val_files=$r1_test_path \
    data.train_batch_size=512 \
    data.max_prompt_length=$max_prompt_length \
    data.max_response_length=$max_response_length \
    data.truncation='left' \
    +actor_rollout_ref.actor.tb_type=tempered_important_sampling \
    +actor_rollout_ref.actor.porj_layer=3 \
    actor_rollout_ref.model.path=$PRETRAINED_MODEL \
    actor_rollout_ref.actor.optim.lr=1e-6 \
    actor_rollout_ref.actor.optim.lr_warmup_steps=10 \
    actor_rollout_ref.actor.optim.weight_decay=0.1 \
    actor_rollout_ref.model.use_remove_padding=True \
    actor_rollout_ref.actor.use_dynamic_bsz=True \
    actor_rollout_ref.ref.log_prob_use_dynamic_bsz=True \
    actor_rollout_ref.rollout.log_prob_use_dynamic_bsz=True \
    actor_rollout_ref.rollout.max_num_batched_tokens=$((max_prompt_length + max_response_length)) \
    actor_rollout_ref.actor.ppo_max_token_len_per_gpu=$((max_prompt_length + max_response_length)) \
    actor_rollout_ref.ref.log_prob_max_token_len_per_gpu=$((max_prompt_length + max_response_length)) \
    actor_rollout_ref.rollout.log_prob_max_token_len_per_gpu=$((max_prompt_length + max_response_length)) \
    actor_rollout_ref.actor.ppo_mini_batch_size=32 \
    actor_rollout_ref.actor.use_kl_loss=True \
    actor_rollout_ref.actor.kl_loss_coef=0.0 \
    actor_rollout_ref.actor.entropy_coeff=0 \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.actor.fsdp_config.param_offload=False \
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=False \
    actor_rollout_ref.rollout.tensor_model_parallel_size=$tensor_model_parallel_size \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.8 \
    actor_rollout_ref.rollout.n=8 \
    actor_rollout_ref.ref.fsdp_config.param_offload=False \
    algorithm.use_kl_in_reward=False \
    trainer.critic_warmup=0 \
    trainer.logger=['console','tensorboard'] \
    trainer.project_name='FlowRL' \
    trainer.experiment_name=$experiment_name \
    trainer.n_gpus_per_node=$n_gpus_per_node \
    trainer.nnodes=$n_nodes \
    trainer.resume_mode=auto \
    trainer.save_freq=$save_freq \
    trainer.default_local_dir=$OUTPUT_DIR \
    trainer.test_freq=5 \
    trainer.total_epochs=1 \
    $@

**Comment by Chenruishuo** [comment]
> Dear Xuekai-Zhu Have you ever tried training Qwen3-base models on this code? I use same bash for Qwen2.5-7B and Qwen3-1.7B. I found that 7B takes only 9 minutes while 1.7B takes 48 minutes. I wonder if there is some speed-up methods designed for Qwen2.5? Here is my training bash:(vllm 0.8.5)
> 
> PRETRAINED_MODEL=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/github.com/volcengine/verl.git/huggingface.co/Qwen/Qwen3-1.7B n_nodes=1 n_gpus_per_node=8 tensor_model_parallel_size=4 save_freq=50
> 
> dapo_train_path=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/flowrl/FlowRL-main/data/dapo-math-17k.parquet r1_test_path=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/flowrl/FlowRL-main/data/validation.parquet
> 
> experiment_name="flowrl_qwen_1.7b_math" max_prompt_length=2048 max_response_length=8192 OUTPUT_DIR=/mnt/dolphinfs/hdd_pool/docker/user/hadoop-mtsearch-assistant/ai-search/linzihan05/Verl_flow/verl-0.5.0/ckpt/$experiment_name
> 
> set -x
> 
> python3 -m verl.trainer.main_ppo algorithm.adv_estimator=grpo data.train_files=$dapo_train_path data.val_files=$r1_test_path data.train_batch_size=512 data.max_prompt_length=$max_prompt_length data.max_response_length=$max_response_length data.truncation='left' +actor_rollout_ref.actor.tb_type=tempered_important_sampling +actor_rollout_ref.actor.porj_layer=3 actor_rollout_ref.model.path=$PRETRAINED_MODEL actor_rollout_ref.actor.optim.lr=1e-6 actor_rollout_ref.actor.optim.lr_warmup_steps=10 actor_rollout_ref.actor.optim.weight_decay=0.1 actor_rollout_ref.model.use_remove_padding=True actor_rollout_ref.actor.use_dynamic_bsz=True actor_rollout_ref.ref.log_prob_use_dynamic_bsz=True actor_rollout_ref.rollout.log_prob_use_dynamic_bsz=True actor_rollout_ref.rollout.max_num_batched_tokens=$((max_prompt_length + max_response_length)) actor_rollout_ref.actor.ppo_max_token_len_per_gpu=$((max_prompt_length + max_response_length)) actor_rollout_ref.ref.log_prob_max_token_len_per_gpu=$((max_prompt_length + max_response_length)) actor_rollout_ref.rollout.log_prob_max_token_len_per_gpu=$((max_prompt_length + max_response_length)) actor_rollout_ref.actor.ppo_mini_batch_size=32 actor_rollout_ref.actor.use_kl_loss=True actor_rollout_ref.actor.kl_loss_coef=0.0 actor_rollout_ref.actor.entropy_coeff=0 actor_rollout_ref.model.enable_gradient_checkpointing=True actor_rollout_ref.actor.fsdp_config.param_offload=False actor_rollout_ref.actor.fsdp_config.optimizer_offload=False actor_rollout_ref.rollout.tensor_model_parallel_size=$tensor_model_parallel_size actor_rollout_ref.rollout.gpu_memory_utilization=0.8 actor_rollout_ref.rollout.n=8 actor_rollout_ref.ref.fsdp_config.param_offload=False algorithm.use_kl_in_reward=False trainer.critic_warmup=0 trainer.logger=['console','tensorboard'] trainer.project_name='FlowRL' trainer.experiment_name=$experiment_name trainer.n_gpus_per_node=$n_gpus_per_node trainer.nnodes=$n_nodes trainer.resume_mode=auto trainer.save_freq=$save_freq trainer.default_local_dir=$OUTPUT_DIR trainer.test_freq=5 trainer.total_epochs=1 $@

Have you noticed that response_length increases a lot? I think it is because that Qwen3 is a reasoning model by default, thus can rollout much longer trajectories. If so, you should pass "enable_thinking=False" to the gen_kwags.

**Comment by 1229095296** [comment]
Thank you so much. That works, the time declines from 48 minutes to 9 minutes. On Qwen3-8B, it takes 20 minutes. My framework is Verl0.5.0, can be referenced for other people. I noticed that verl0.6.0 includes the FlowRL naturally. I will try it later.

**Comment by Xuekai-Zhu** [comment]
> Thank you so much. That works, the time declines from 48 minutes to 9 minutes. On Qwen3-8B, it takes 20 minutes. My framework is Verl0.5.0, can be referenced for other people. I noticed that verl0.6.0 includes the FlowRL naturally. I will try it later.

If you want to conduct code experiments, official verl has no implementation for code reward. The code reward is sourced from：https://github.com/rllm-org/rllm. 

If you're trying to use the FlowRL recipe in verl, I strongly suggest using the latest version, as there are some training/inference mismatching issues in earlier versions.

**Comment by karenyun** [comment]
Hi @1229095296 @Xuekai-Zhu , I have the same problem when used the `Qwen3-1.7B`, the time for every step is about 1 hour. 

Howerver, when I add this parameters `+actor_rollout_ref.rollout.enable_thinking=False,` the time dose not change .
Could you share your setting? Thanks so much!

**Comment by Chenruishuo** [comment]
> Hi [@1229095296](https://github.com/1229095296) [@Xuekai-Zhu](https://github.com/Xuekai-Zhu) , I have the same problem when used the `Qwen3-1.7B`, the time for every step is about 1 hour.
> 
> Howerver, when I add this parameters `+actor_rollout_ref.rollout.enable_thinking=False,` the time dose not change . Could you share your setting? Thanks so much!

It seems that you must directly pass the "enable_thinking" parameter in code line "tokenizer.apply_chat_template(...,enable_thinking=False" or add "/no_think" at the end of your prompt. You can check [Qwen3](https://huggingface.co/Qwen/Qwen3-4B) for more detail.
